### PR TITLE
Swift 3.0

### PR DESCRIPTION
**Pull request by ikesyo**
This succeeds the @norio-nomura's work of #277.

### CODE REVIEW

**Comment by jeffh** [comment]
@ikesyo, what's left that prevents this from merging into master?

**Comment by ikesyo** [comment]
@jeffh I think this should get additional reviews, for example:

- API Design (Naming) Guidelines
- `IPHONEOS_DEPLOYMENT_TARGET` is bumped to `8.0`
- `BeIdenticalTo` implementation which is affected by id-as-Any.

File: Sources/Nimble/DSL.swift [hunk 1]
```diff
@@ -58,11 +56,10 @@ internal func nimblePrecondition(
         return result
```
Comment (jeffh) [inline Sources/Nimble/DSL.swift]: We should get rid of the return value because it isn't used.

**Comment by jeffh** [comment]
Thanks for the update @ikesyo.

> API Design (Naming) Guidelines

I think that can be addressed when we attempt to switch matchers to use protocol extensions. I'll try again next week to play around and see if protocol extensions don't cause all sorts of problems.

> `IPHONEOS_DEPLOYMENT_TARGET` is bumped to `8.0` (we can't select `7.0` on Xcode 8 GUI anymore)

👍 . This is fine. If someone still manages to get it to work on 7.0, we can accept a PR as needed.

> `BeIdenticalTo` implementation which is affected by id-as-Any.

While not ideal, the implementation seems workable.

Other than that one change. It seems ready to merge, imo.

**Comment by ikesyo** [comment]
@jeffh Thanks for the review! I've addressed the return value of `nimblePrecondition`.

**Comment by jeffh** [comment]
Thanks everyone for putting in all the effort! 👏

**Comment by tettoffensive** [comment]
Even though I have specified in my Podfile, Xcode 8 GM still complains and wants to convert to swift 3 for both projects. am I missing something?

```bash
pod 'Quick', :git => 'https://github.com/Quick/Quick.git', :branch => 'swift-3.0'
pod 'Nimble', :git => 'https://github.com/Quick/Nimble.git', :branch => 'master'
```

**Comment by tettoffensive** [comment]
@jeffh @ikesyo ?

**Comment by ikesyo** [comment]
You might currently use CocoaPods 1.0.1, but you need 1.1.0.rc.2 for Xcode 8: `gem update cocoapods --pre`

**Pull request merged**